applicationset/generators/generator_spec_processor.go

```diff
@@ -31,7 +33,17 @@ func Transform(requestedGenerator argoprojiov1alpha1.ApplicationSetGenerator, al
 			}
 			continue
 		}
-
+		if len(genParams) != 0 {
```

Comment:
unless g.GenerateParams is called as in line no 47, genParams wouldn't be always zero?

Reply:
Not for the Matrix generator. It will be 0 when called from the applicationset_controller.go file, but when it's called from matrix.go I pass in the previous generator's params.